compute_tools/src/configurator.rs

```diff
@@ -42,13 +42,15 @@ fn configurator_main_loop(compute: &Arc<ComputeNode>) {
     }
 }
 
-pub fn launch_configurator(compute: &Arc<ComputeNode>) -> Result<thread::JoinHandle<()>> {
+pub fn launch_configurator(
+    compute: &Arc<ComputeNode>,
+) -> Result<thread::JoinHandle<()>, std::io::Error> {
```

Comment:
It might be a good time to remove the `use anyhow::Result;` though... I cannot see why one would make this change without any error handling changes elsewhere.

`anyhow::Error` is nice in terms of it only costing a single usize of storage, which with custom std::io::Error could be much more. Here however with thread spawning I guess more natural is to just unwrap for the rare case of not being allowed to create more threads.

Reply:
yeah it just gets unwrapped in `compute_ctl` later

Reply:
@koivunej let me know if you like the new version, or think that this is not worth changing.

Reply:
It's looking good; I added a comment to `std::thread::spawn` impl. So this builder does expose the failure to create a thread, but we should never see those in practice. If we would, then it would be very interesting and warrant a panic :)

Reply:
Actually: please update the description (and possibly squash merge message).